andela/ah-backend-tesseract
# #159951982 Enable bookmarking of articles

### What does this PR?

Implements endpoints for creating, viewing and deleting bookmarks.

### Description of Task to be completed?

Have the following endpoints working:

`POST /api/article/:slug/bookmark`

`GET /api/articles/bookmarks`

`DELETE /api/article/:slug/bookmark/delete`

### How should this be manually tested?
Run the project, create an article and navigate to each of the above endpoints. 

Note that the `POST /api/article/:slug/bookmark` endpoint doesn't require anything as part of the body of the request.

### What are the relevant pivotal tracker stories?

#159951982

### Screenshots
Response when someone creates a bookmark:
<img width="509" alt="screen shot 2018-09-19 at 17 31 31" src="https://user-images.githubusercontent.com/26762336/45760143-14506580-bc32-11e8-9909-f61c33df2917.png">

Viewing bookmarks:
<img width="681" alt="screen shot 2018-09-19 at 17 31 59" src="https://user-images.githubusercontent.com/26762336/45760194-321dca80-bc32-11e8-8660-339eb5a9d041.png">

Response when someone deletes a bookmark:
<img width="433" alt="screen shot 2018-09-19 at 17 32 12" src="https://user-images.githubusercontent.com/26762336/45760234-45c93100-bc32-11e8-976b-4a28d11b0e5f.png">
